src/rascal/representations/cutoff_functions.hh

```diff
@@ -229,28 +230,28 @@ namespace rascal {
 
       double value(double distance) {
         double factor{0.};
-        if (this->rate > math::DBL_FTOL) {
-          factor = this->rate / (this->rate + math::pow(distance / this->scale,
-                                                        this->exponent));
+        if (std::abs(this->rate) <= math::DBL_FTOL) {
+          factor = 1. / math::pow(distance / this->scale, this->exponent);
         } else if (this->exponent == 0) {
           factor = 1.;
         } else {
-          factor = 1. / math::pow(distance / this->scale, this->exponent);
+          factor = this->rate / (this->rate + math::pow(distance / this->scale,
```

Comment:
There might be a small gain to have this case, the most likely, first in the if branch. 
this->rate and  this->exponent don't change after initialization so you could check for these 3 cases there and set an intermediate variable to fit any of the 3 cases with this one being first ?

Reply:
do you really think we should try to outsmart branch prediction, without any benchmarks, with a small loss in readability?

Reply:
when using the radial scaling this is the main use case. Even if the branching is properly predicted 2 comparison are evaluated in any cases with this order so it would not be as good as the other way around.